include/swift/IRGen/IRGenPublic.h

```diff
@@ -26,7 +26,8 @@ class IRGenModule;
 
 /// Create an IRGen module.
 std::pair<IRGenerator *, IRGenModule *>
-createIRGenModule(SILModule *SILMod, llvm::LLVMContext &LLVMContext);
+createIRGenModule(SILModule *SILMod, PrimarySpecificPaths PSPs,
```

Comment:
Seems like this should be passed by const reference (here and elsewhere in IRGen/Subsystems). It's not cheap to copy.

Reply:
Copying avoids reliance on FrontendInputsAndOutputs not getting freed. Do you think that saving cost of copying perhaps a dozen strings is worth an implicit lifetime dependency?

Reply:
Or is the dependency already explicit?

Reply:
It's sort of explicit? SILModule -> ASTContext -> DiagnosticEngine -> CompilerInstance. But mostly we just assume it's better *not* to copy something when passing information down the stack—almost all lifetimes are stack-oriented. If you do a copy anywhere, it would be at the point where you move the PSPs into the IRGenModule, because that gets persisted beyond the constructor call.

Reply:
OK, will fix createIRGenModule to take and pass down a const ref for PSPs.

Reply:
Thanks for explaining.

Reply:
I have made some changes to pass down const refs.